### ISSUE TITLE
clickhouse jdbc 通过server写入数据，不会将任务分发给worker，直接连接worker，又会报错

### ISSUE THREAD

**Issue by yijiluo** [ByConity/ByConity#1594]
(you don't have to strictly follow this form)

## Enhancement
clickhouse jdbc 通过server写入数据（insert values），不会将任务分发给worker，但是直接连接worker写入，clickhouse jdbc 默认会先对写入表执行select语句，此时会报错：DB::Exception: Worker group is not set，因为 worker write不能执行read操作。
![image](https://github.com/ByConity/ByConity/assets/131134513/f967e586-8a6c-4500-a5f8-3460de6025b9)

**Comment by yijiluo** [comment]
也可能是spark处理jdbc写入的时候，发了一个查询请求，发到 worker write，就会报错，但是发送insert values请求，就是正常的

**Comment by yijiluo** [comment]
![image](https://github.com/ByConity/ByConity/assets/131134513/210ff6b7-0106-4b21-965d-0e45074f006a)

**Comment by yijiluo** [comment]
![image](https://github.com/ByConity/ByConity/assets/131134513/8c3612c8-4e5a-48c0-8191-34311f968749)
clickhosue jdbc 获取列信息的时候，也会报错

**Comment by nudles** [comment]
> (you don't have to strictly follow this form)
> 
> ## Enhancement
> clickhouse jdbc 通过server写入数据（insert values），不会将任务分发给worker，但是直接连接worker写入，clickhouse jdbc 默认会先对写入表执行select语句，此时会报错：DB::Exception: Worker group is not set，因为 worker write不能执行read操作。 ![image](https://private-user-images.githubusercontent.com/131134513/320256960-f967e586-8a6c-4500-a5f8-3460de6025b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NDI2NzgsIm5iZiI6MTcxMjU0MjM3OCwicGF0aCI6Ii8xMzExMzQ1MTMvMzIwMjU2OTYwLWY5NjdlNTg2LThhNmMtNDUwMC1hNWY4LTM0NjBkZTYwMjViOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQwMjEyNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzE5NTVlMzY1NDI4OTgyYzUxOTcwMzFiYzU0ZjZjYTVmMjgxODViZDcyMGYzNDE4ZDVlMmEzNWVlMzU5MGE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pfJBR5VnUTwLPoB3L9kVATwmHLyxqMdncRmREpJaXRw)

@yijiluo what's your cluster configuration? 1 server + x default workers + y write workers?
Do you mean that the server never forwards the insert requests to the write workers?

**Comment by yijiluo** [comment]
> > (you don't have to strictly follow this form)
> > ## Enhancement
> > clickhouse jdbc 通过server写入数据（insert values），不会将任务分发给worker，但是直接连接worker写入，clickhouse jdbc 默认会先对写入表执行select语句，此时会报错：DB::Exception: Worker group is not set，因为 worker write不能执行read操作。 ![image](https://private-user-images.githubusercontent.com/131134513/320256960-f967e586-8a6c-4500-a5f8-3460de6025b9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NDI2NzgsIm5iZiI6MTcxMjU0MjM3OCwicGF0aCI6Ii8xMzExMzQ1MTMvMzIwMjU2OTYwLWY5NjdlNTg2LThhNmMtNDUwMC1hNWY4LTM0NjBkZTYwMjViOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQwMjEyNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzE5NTVlMzY1NDI4OTgyYzUxOTcwMzFiYzU0ZjZjYTVmMjgxODViZDcyMGYzNDE4ZDVlMmEzNWVlMzU5MGE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pfJBR5VnUTwLPoB3L9kVATwmHLyxqMdncRmREpJaXRw)
> 
> @yijiluo what's your cluster configuration? 1 server + x default workers + y write workers? Do you mean that the server never forwards the insert requests to the write workers?

yes，my cluster is 1 server + 2 default workers + 3 write workers.
If I use clickhouse jdbc to connect to byconity, and send insert values requests, the server CPU load and disk IO wait will be high, and woker write will not have much load. This issue has been reported in the wechat community group and has been replied to. Currently, insert values requests will not be executed by woker write
![image](https://github.com/ByConity/ByConity/assets/131134513/245838a6-787c-4559-8f80-bc8e3a3563e8)

**Comment by skyoct** [comment]
@yijiluo Hi, You can refer to #722.

**Comment by yijiluo** [comment]
> @yijiluo Hi, You can refer to #722.

Yeah, I got it. I set <prefer_cnch_catalog>1</prefer_cnch_catalog> in user.xml on woker write, and i can write by http, when I use clickhouse jdbc to write , it send select to worker write and report error, please see it.
https://private-user-images.githubusercontent.com/131134513/320262064-8c3612c8-4e5a-48c0-8191-34311f968749.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NjkzNzYsIm5iZiI6MTcxMjU2OTA3NiwicGF0aCI6Ii8xMzExMzQ1MTMvMzIwMjYyMDY0LThjMzYxMmM4LTRlNWEtNDhjMC04MTkxLTM0MzExZjk2ODc0OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOFQwOTM3NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MmEyMGJkYTkyMjQwYmY0Y2FlYTdmMDUwMDYyYzYzNDZhZDlhNzdlZjc5Mjg3NGJjY2IxMzVlYWZhMjgwNDg2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yBAKgba7v5a9aYSXtthgGW6x_h9yhk9HWAQPT1kQgXA
